garage/tf/policies/gaussian_mlp_policy.py

```diff
@@ -47,214 +47,89 @@ def __init__(self,
-                    with tf.variable_scope(self._std_network_name):
-                        l_std_param = L.SliceLayer(
-                            mean_network.output_layer,
-                            slice(action_dim, 2 * action_dim),
-                            name="std_slice",
-                        )
-                else:
-                    with tf.variable_scope(self._std_network_name):
-                        l_std_param = L.ParamLayer(
-                            mean_network.input_layer,
-                            num_units=action_dim,
-                            param=tf.constant_initializer(init_std_param),
-                            name="output_std_param",
-                            trainable=learn_std,
-                        )
-
-            self.std_parametrization = std_parametrization
-
-            if std_parametrization == 'exp':
-                min_std_param = np.log(min_std)
-            elif std_parametrization == 'softplus':
-                min_std_param = np.log(np.exp(min_std) - 1)
-            else:
-                raise NotImplementedError
-
-            self.min_std_param = min_std_param
-
-            # mean_var, log_std_var = L.get_output([l_mean, l_std_param])
-            #
-            # if self.min_std_param is not None:
-            #     log_std_var = tf.maximum(log_std_var, np.log(min_std))
-            #
-            # self._mean_var, self._log_std_var = mean_var, log_std_var
-
-            self._l_mean = l_mean
-            self._l_std_param = l_std_param
-
-            self._dist = DiagonalGaussian(action_dim)
-
-            LayersPowered.__init__(self, [l_mean, l_std_param])
-            super(GaussianMLPPolicy, self).__init__(env_spec)
-
-            dist_info_sym = self.dist_info_sym(
-                mean_network.input_layer.input_var, dict())
-            mean_var = tf.identity(dist_info_sym["mean"], name="mean")
-            log_std_var = tf.identity(
-                dist_info_sym["log_std"], name="standard_dev")
-
-            self._f_dist = tensor_utils.compile_function(
-                inputs=[obs_var],
-                outputs=[mean_var, log_std_var],
-            )
+        
+        self.model = GaussianMLPModel(
+            input_dim=env_spec.observation_space.flat_dim,
+            output_dim=env_spec.action_space.flat_dim,
+            hidden_sizes=hidden_sizes,
+            learn_std=learn_std,
+            init_std=init_std,
+            adaptive_std=adaptive_std,
+            std_share_network=std_share_network,
+            std_hidden_sizes=std_hidden_sizes,
+            min_std=min_std,
+            max_std=max_std,
+            std_hidden_nonlinearity=std_hidden_nonlinearity,
+            hidden_nonlinearity=hidden_nonlinearity,
+            output_nonlinearity=output_nonlinearity,
+            std_parameterization=std_parameterization,
+        )
+
+        self._dist = DiagonalGaussian(self.env_spec.action_space.flat_dim)
```

Comment:
Why not using self.model["dist"]?

Reply:
Because of the current implementation of NPO (which I don't want to fix for now).

Reply:
Are you going to do benchmarking for the new GaussianMLPPolicy? If you are going to benchmark test the policy on algorithms, you need to use NPO/PPO/TRPO as they are the algorithms that use GaussianMLPPolicy for now.